### PR TITLE
feat: migrate docs listicles to CMS

### PR DESCRIPTION
**Pull request by manika-signoz**
## Pull Request

---

#### Screenshots / Screen Recordings (if applicable)

TBA

All pages render identically to production.

#### Issues closed by this PR

N/A

---

### ✅ Change Type
_Select all that apply_

- [x] ✨ Feature
- [x] 🛠️ Infra / Tooling
- [x] 🧪 Test-only

---

- Tests added/updated:
  - `tests/component-items-sync.test.js` - 6 tests covering JSON parsing, schema validation, item validation, icon asset existence in `data-assets/`, and duplicate href detection
  - `tests/listicle-cms-transform.test.js` - CMS transform round-trip tests
- Manual verification:
  - Visual comparison of 7 preview pages vs production (all listicle patterns: flat, sectioned, searchable)
  - Tab switching and search filtering verified on preview
- Edge cases covered: NA

---

- Blast radius: Docs pages that render `<Listicle>` components (28 pages). No impact on blog, comparisons, guides, or other content types.
- Potential regressions: If Strapi CMS is down or returns invalid data, the listicle component renders an error message. CMS is the sole source of truth in production (matches content repository pattern).
- Rollback plan: Revert PR

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested
- [x] Breaking changes documented
- [x] Backward compatibility considered

---

## 👀 Notes for Reviewers
Stacks before https://github.com/SigNoz/signoz.io/pull/3504

### CODE REVIEW

File: utils/listicleRepository.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    return getLocalListicleConfig(name)
+  }
+
+  try {
+    return await getCachedCmsListicle(name)
+  } catch (error) {
+    console.error(`Failed to fetch listicle "${name}" from CMS:`, error)
+    return null
+  }
+}
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply utils/listicleRepository.ts:141]: the listicles are cached on cdn using unstable_cache, so not all requests would directly go through to the cms, if cache is stale + cms down will we run into this issue, accepted risk

File: scripts/sync-content-to-strapi.js [hunk 2]
```diff
@@ -312,6 +322,135 @@ function parseMDXFile(filePath) {
...
+
+  items.forEach((item, index) => {
+    assertNonEmptyString(item.name, `${fieldName}[${index}].name`)
+    assertValidHref(item.href, `${fieldName}[${index}].href`)
+    assertValidIcon(item.icon, `${fieldName}[${index}].icon`)
+    if (item.clickName !== undefined) {
+      assertNonEmptyString(item.clickName, `${fieldName}[${index}].clickName`)
+    }
+  })
+}
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply scripts/sync-content-to-strapi.js:373]: all listicles must have icons, ts type should be fixed.

File: utils/listicleCms.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+  if (item.icon_path) return item.icon_path
+  if (item.icon_badge && item.icon_color) {
+    return {
+      badge: item.icon_badge,
+      color: item.icon_color,
+    }
+  }
+
+  return undefined
+}
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply utils/listicleCms.ts:87]: added console.warn when only one of badge/color is present, fallback to empty string for strict type checking

File: tests/component-items-sync.test.js [hunk 4]
```diff
@@ -231,6 +231,23 @@ test('all listicle items have valid name, href, and icon', () => {
...
+        const cleanPath = item.icon.startsWith('/') ? item.icon.slice(1) : item.icon
+        const dataAssetsPath = join(__dirname, '..', 'data-assets', cleanPath)
+        assert.ok(
+          require('fs').existsSync(dataAssetsPath),
+          `${name} item "${item.name}" icon "${item.icon}" missing from data-assets/`
+        )
+      }
+    }
+  }
+})
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply tests/component-items-sync.test.js:249]: done

File: scripts/check-cms-assets.js [hunk 5]
```diff
@@ -90,6 +91,32 @@ function getStagedMigratedFiles() {
     .filter((f) => f && MIGRATED_PATTERN.test(f))
 }
 
+function getStagedListicleFiles() {
+  const output = execSync('git diff --cached --name-only --diff-filter=ACM', { encoding: 'utf8' })
+  return output
+    .trim()
+    .split('\n')
+    .filter((f) => f && LISTICLE_PATTERN.test(f))
+}
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply scripts/check-cms-assets.js]: done

**Review by H4ad — APPROVED** [review]

**Comment by manika-signoz** [comment]
closing in favour of https://github.com/SigNoz/signoz.io/pull/3578
